tests/pkg_fatfs_vfs/main.c

```diff
@@ -0,0 +1,300 @@
+/*
+ * Copyright (C) 2017 HAW-Hamburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     tests
+ * @{
+ *
+ * @file
+ * @brief       test application for using fatfs with vfs
+ *
+ * @author      Michel Rottleuthner <michel.rottleuthner@haw-hamburg.de>
+ * @}
+ */
+
+#include <stdio.h>
+#include <fcntl.h>
+#include <string.h>
+#include <errno.h>
+
+#include "fs/fatfs.h"
+#include "vfs.h"
+#include "fs/fatfs.h"
+
+#if FATFS_FFCONF_OPT_FS_NORTC == 0
+#include "periph/rtc.h"
+#endif
+
+#define MNT_PATH  "/test"
+#define FNAME1 "TEST.TXT"
+#define FNAME2 "NEWFILE.TXT"
+#define FNAME_RNMD  "RENAMED.TXT"
+#define FNAME_NXIST "NOFILE.TXT"
+#define FULL_FNAME1 (MNT_PATH "/" FNAME1)
+#define FULL_FNAME2 (MNT_PATH "/" FNAME2)
+#define FULL_FNAME_RNMD  (MNT_PATH "/" FNAME_RNMD)
+#define FULL_FNAME_NXIST (MNT_PATH "/" FNAME_NXIST)
+#define DIR_NAME "SOMEDIR"
+
+static const char test_txt[]   = "the test file content 123 abc";
+static const char test_txt2[] = "another text";
+static const char test_txt3[] = "hello world for vfs";
+
+static fatfs_desc_t fatfs = {
+    .fat_fs = { 0 },
+    .vol_idx = 0
+};
+
+static vfs_mount_t _test_vfs_mount = {
+    .mount_point = MNT_PATH,
+    .fs = &fatfs_file_system,
+    .private_data = (void *)&fatfs,
+};
+
+static void print_test_result(const char *test_name, int ok)
+{
+    printf("%s:[%s]\n", test_name, ok ? "OK" : "FAILED");
+}
+
+static void test_mount(void)
+{
+    print_test_result("test_mount__mount", vfs_mount(&_test_vfs_mount) == 0);
```

Comment:
all the pexpect-based tests I looked at are using print. See for [tests/xtimer_usleep](https://github.com/RIOT-OS/RIOT/blob/master/tests/xtimer_usleep/main.c#L38) for example. This test is not located in tests/unittests/.